extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -87,12 +81,27 @@ public Object get(ByteBuffer buf, int position)
   //Note that this is not threadsafe and I don't think it needs to be
   private Union getUnion(ByteBuffer buf, int position)
   {
-    Union union = unions.get(position);
-    if (union == null) {
-      Memory mem = new MemoryRegion(nm, position, maxIntermediateSize);
-      union = (Union) SetOperation.wrap(mem);
-      unions.put(position, union);
+    Int2ObjectMap<Union> unionMap = unions.get(buf);
+    Union union = unionMap != null ? unionMap.get(position) : null;
+    if (union != null) {
+      return union;
+    }
+    return createNewUnion(buf, position, true);
+  }
+
+  private Union createNewUnion(ByteBuffer buf, int position, boolean isWrapped)
+  {
+    NativeMemory nm = getNativeMemory(buf);
+    Memory mem = new MemoryRegion(nm, position, maxIntermediateSize);
+    Union union = isWrapped
+                  ? (Union) SetOperation.wrap(mem)
+                  : (Union) SetOperation.builder().initMemory(mem).build(size, Family.UNION);
+    Int2ObjectMap<Union> unionMap = unions.get(buf);
```

Comment:
@akashdw have you even tried to analyze the results? Or you supposed that I should do this for you?

 - This benchmark is doing computeIfAbsent() against `Int2ObjectOpenHashMap` rather than   `IdentityHashMap`, like in the PR
 - Default `@Setup` level is `Level.Trial`, that makes little sense for such code. Should be `Level.Invocation`.
 - The actual reason of the difference, is that `Int2ObjectOpenHashMap::new` takes the key as capacity of the map. It is a bug in the code that I suggested. It should be `k -> new Int2ObjectOpenHashMap()`.

So the results of this benchmark: https://gist.github.com/leventov/2dabf3563e50012493e030aa69a40be4
```
ComputeIfAbsent.withComputeIfAbsent     avgt   10  83.784 ± 1.476  us/op
ComputeIfAbsent.withoutComputeIfAbsent  avgt   10  85.201 ± 1.083  us/op
```

I. e. versions perform identically, that is just as expected, because `IdentityHashMap` actually doesn't override `computeIfAbsent()`. The default implementation of computeIfAbsent() does exactly the same calls as you do in the hand-written version. However
 - `IdentityHashMap` may specialize `computeIfAbsent()` in the future versions of JDK;
 - At some point of time `unions` and `nmCache` could be changed to use e. g. `HashMap`, which specializes `computeIfAbsent()`, and the code that already uses `computeIfAbsent()` will automatically improve;
 - All other advantages of `computeIfAbsent()` are still in place.

<hr>

Note: `-server` is the default option with JDK 8, you don't need to specify it explicitly.